### PR TITLE
Document toolchain task feature

### PR DESCRIPTION
**Pull request by bmuskalla**
Document the new task and add release notes.

Rendered version:
Screenshot 2020-10-21 at 15 11 35

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc [hunk 1]
```diff
@@ -142,6 +142,42 @@ Only secure protocols like `https` are accepted.
 This is required to make sure no one can tamper with the download in flight.
 ====
 
+[[sub:viewing_toolchains]]
+=== Viewing and debugging toolchains
+
+Gradle can visualize the list of all detected toolchains including their metadata.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:148]: > can display

visualize here does not work for me

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc [hunk 2]
```diff
@@ -142,6 +142,42 @@ Only secure protocols like `https` are accepted.
...
+     | Detected by:        SDKMAN!
+
+ + Oracle JDK 1.7.0_80
+     | Location:           /Library/Java/JavaVirtualMachines/jdk1.7.0_80.jdk/Contents/Home/jre
+     | Language Version:   7
+     | Is JDK:             true
+     | Detected by:        macOS java_home
+----
+
+This can help to debug which toolchains are available to the build, how they are detected and what kind of metadata Gradle knows about those toolchains.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:179]: Maybe you could describe what is displayed for toolchains that are passed through env vars or references to their java home directly?

**Review by ljacomet — COMMENTED** [review]
Suggested improvements, see below

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc (hunk 2 above)
Comment (bmuskalla) [reply subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:179]: Good idea! Done

**Review by ljacomet — APPROVED** [review]

**Comment by pioterj** [comment]
@bmuskalla I'd suggest adding a bit of context at the beginning. Specifically:

- Start by briefly mentioning the toolchain support and what it does as it's a relatively new feature and many people won't recognize it yet. For example "Java toolchain support provides an easy way to declare what Java version the project should be built with".  
- Then express the problem - Why you may need this list?

**Pull request merged**